src/main/java/io/vertx/core/file/impl/FileResolver.java

```diff
@@ -398,18 +408,20 @@ private synchronized void setupCacheDir(String id) {
       }
     });
     Runtime.getRuntime().addShutdownHook(shutdownHook);
+    return cacheDir;
   }
 
   private void deleteCacheDir() throws IOException {
-    Path path;
+    if (cacheDir == null || !cacheDir.exists()) {
```

Comment:
instead here we should have something that will not block other threads when deleting the dir and just return, something like

```java
File dir;
synchronized(this) {
  if (cacheDir == null) {
    return;
  }
  dir = cacheDir;
  cacheDir = null;
}
if (dir.exists()) {
  FileSystemImpl.delete(dir.toPath(), true);
}
```